src/gui/viewer.cpp

```diff
@@ -390,6 +393,50 @@ void ViewerMainWindow::reload_file() {
 	document()->load_file(document()->path().c_str());
 }
 
+void ViewerMainWindow::import_file() {
+	auto filename = QFileDialog::getOpenFileName(this, tr("Open Image"), QString(), "JPG Files (*.jpg)").toUtf8();
+
+	if (filename.isEmpty())
+		return;
+
+	auto image = imglib::image_begin(filename);
+
+	if (!image) {
+		QMessageBox::warning(
+			this, "File does not exist", fmt::format(FMT_STRING("Failed to open image: {}"), filename.data()).c_str(),
+			QMessageBox::Ok);
+		return;
+	}
+
+	auto data = imglib::image_load(image);
+
+	auto comps = data.info.comps;
+	auto w = data.info.w;
+	auto h = data.info.h;
```

Comment:
You need to null check `data.data` too- forgot about that! That'll end up being null if the image file could not be loaded (ie if it's corrupted or something)